pokemon-showdown

```diff
@@ -22,6 +24,84 @@ try {
 	child_process.execSync('npm install --production', {stdio: 'inherit'});
 }
 
+// Check if the server is configured to use Go, and ensure the required
+// environment variables and dependencies are available if that is the case
+
+let config;
+try {
+	config = require('./config/config');
+} catch (e) {}
+
+if (config && config.golang) {
+	if (!process.env.GOPATH) {
+		console.log('The GOPATH environment variable is not set! It is required in order to run the server using Go.');
+		process.exit(0);
+	}
+	if (!process.env.GOROOT) {
```

Comment:
GOROOT is optional, and probably shouldn't be set, as its default should be fine. Don't require it, it's probably a bad idea to set that variable explicitly anyway.